### PR TITLE
[Platform] Enable L10n linter on new betas and releases

### PR DESCRIPTION
**Pull request by AliSoftware**
# Context

This PR enables the linting of localizations by calling the action (previously introduced by https://github.com/wordpress-mobile/release-toolkit/pull/174).

References: paaHJt-D6-p2 & paNNhX-am-p2 (Step 1)

# What does it do?

This PR will call the linter right after the we download the updated translations from GlotPress, which is every time we do a `finalize_release`, but now also when we do a `new_beta_release` (which didn't update the translations up until now in WC unlike it does in WP, which is now fixed too to help us catch L10n violations ASAP during the beta-testing cycle before final release)

If any violations are found, the lane will stop and prevent us from going further (as those violations would otherwise make the app crash in those problematic locales. In those cases, the release manager is supposed to reject the problematic strings in GlotPress (or fix them themselves if they are fluent in that locale and the fix is straightforward) before triggering the `new_beta_release` or `finalize_release` lanes again.

# PR Requirements before we merge

* This PR requires that https://github.com/wordpress-mobile/release-toolkit/pull/177 gets merged first and that we then create a new version of the release-toolkit to point to it in this PR

# Current Status

As of today, all existing violations that existed before I ran the linter has been fixed 🎊 

<details><summary>Result of running the action on the current branch</summary>

```
 $ fl run ios_lint_localizations input_dir:WooCommerce/Resources
[✔] 🚀 
/Users/olivier/Documents/Dev/woocommerce-ios/vendor/bundle/ruby/2.6.0/gems/activesupport-4.2.11.1/lib/active_support/core_ext/object/duplicable.rb:111: warning: BigDecimal.new is deprecated; use BigDecimal() method instead.
+-----------------------------------+---------+------------------------------------------------------------------------+
|                                                     Used plugins                                                     |
+-----------------------------------+---------+------------------------------------------------------------------------+
| Plugin                            | Version | Action                                                                 |
+-----------------------------------+---------+------------------------------------------------------------------------+
| fastlane-plugin-wpmreleasetoolkit | 0.10.0  | configure_download configure_setup configure_apply configure_update    |
|                                   |         | configure_add_files_to_copy configure_validate ios_build_prechecks     |
|                                   |         | ios_betabuild_prechecks ios_build_preflight ios_localize_project       |
|                                   |         | ios_update_release_notes add_devices_to_provisioning_profiles          |
|                                   |         | ios_current_branch_is_hotfix                                           |
|                                   |         | add_development_certificates_to_provisioning_profiles                  |
|                                   |         | ios_merge_translators_strings ios_hotifx_prechecks ios_tag_build       |
|                                   |         | ios_final_tag ios_finalize_prechecks ios_bump_version_hotfix           |
|                                   |         | ios_check_beta_deps ios_validate_ci_build ios_get_app_version          |
|                                   |         | ios_lint_localizations ios_codefreeze_prechecks ios_update_metadata    |
|                                   |         | ios_clear_intermediate_tags ios_bump_version_beta                      |
|                                   |         | ios_completecodefreeze_prechecks ios_update_metadata_source            |
|                                   |         | ios_bump_version_release ios_get_store_app_sizes                       |
|                                   |         | extract_release_notes_for_version gp_downloadmetadata create_release   |
|                                   |         | setfrozentag close_milestone promo_screenshots removebranchprotection  |
|                                   |         | gp_update_metadata_source get_prs_list create_new_milestone            |
|                                   |         | setbranchprotection android_completecodefreeze_prechecks               |
|                                   |         | android_betabuild_prechecks android_bump_version_release               |
|                                   |         | android_current_branch_is_hotfix android_build_prechecks               |
|                                   |         | android_bump_version_final_release android_build_preflight             |
|                                   |         | android_get_release_version android_codefreeze_prechecks               |
|                                   |         | android_tag_build android_create_xml_release_notes                     |
|                                   |         | android_hotifx_prechecks an_localize_libs android_bump_version_hotfix  |
|                                   |         | android_update_release_notes android_update_metadata                   |
|                                   |         | an_validate_lib_strings android_get_alpha_version                      |
|                                   |         | android_get_app_version android_merge_translators_strings              |
|                                   |         | android_finalize_prechecks an_update_metadata_source                   |
|                                   |         | android_bump_version_beta                                              |
| fastlane-plugin-sentry            | 1.6.0   | sentry_upload_dsym sentry_set_commits sentry_upload_sourcemap          |
|                                   |         | sentry_finalize_release sentry_upload_file sentry_upload_proguard      |
|                                   |         | sentry_create_release                                                  |
| fastlane-plugin-appcenter         | 1.6.0   | appcenter_fetch_devices appcenter_upload                               |
| fastlane-plugin-test_center       | 3.10.2  | suppressed_tests suppress_tests suppress_tests_from_junit              |
|                                   |         | tests_from_xctestrun quit_core_simulator_service                       |
|                                   |         | collate_test_result_bundles tests_from_junit multi_scan                |
|                                   |         | collate_html_reports collate_junit_reports collate_json_reports        |
|                                   |         | collate_xcresults                                                      |
+-----------------------------------+---------+------------------------------------------------------------------------+

[18:09:19]: ------------------------------------
[18:09:19]: --- Step: ios_lint_localizations ---
[18:09:19]: ------------------------------------
[18:09:19]: Linting localizations for parameter placeholders consistency...
[18:09:19]: $ /Users/olivier/Documents/Dev/woocommerce-ios/vendor/swiftgen/6.4.0/bin/swiftgen config run --config /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/swiftgen-stringtypes.yml
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.it.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.tr.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.fr.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.id.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.sv.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.ja.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.ar.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.pt-BR.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.he.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.nl.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.ru.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.ko.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.es.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.en.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.de.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.zh-Hans.txt
[18:09:20]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-91872-xq3hqo/L10nParamsList.zh-Hant.txt
[18:09:20]: Result: {}
```
</details>

### CODE REVIEW

File: fastlane/Pluginfile
Comment (AliSoftware) [inline fastlane/Pluginfile]: ⚠️ This needs to be reverted to point to a tag once wordpress-mobile/release-toolkit#177 is merged and we tag a new version of the release toolkit

File: fastlane/Fastfile
Comment (jkmassel) [inline fastlane/Fastfile:213]: Nice

**Review by jkmassel — APPROVED** [review]
LGTM pending the release toolkit tag

File: fastlane/Pluginfile
Comment (mokagio) [reply fastlane/Pluginfile]: image

Neat commit message reminder! I usually write "`REVERT ME`", but I like the extra color here 🚨 .

Reading this made me think "we should have a Peril rule to remind us to revert commits that start with <agreed upon revert pattern>". Then I thought a better option would be to have a Peril rule that fails the build (preventing merging) if the plugin points to anything other than a tag.

It's a reminder that we all set for ourselves when opening these PRs and that reviewers point out, too. Seems ripe for automation.

**Review by mokagio — APPROVED** [review]

**Comment by AliSoftware** [comment]
@jkmassel @mokagio I've updated the PR:
 - Rebased on top of `release/5.6`, updated the PR's target branch accordingly
 - Pointed the release-toolkit to the recently created new tag 0.11.0
 - Update the behavior for `new_beta_release` to allow bypassing the violations for betas (while still strongly suggesting to abort on violations for beta, and while still making it a non-bypass-able error for `finalize_release`

I'll let you review (especially @jkmassel if you have any feedback once you've tested the [similar PR in WordPress-iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/15423) in practice with a new WP-iOS beta you plan to make and might have feedback from actually using it in real life), and if you're ok with the new changes, feel free to merge on green yourself.

File: WooCommerce/WooCommerce.xcodeproj/project.pbxproj
Comment (AliSoftware) [inline WooCommerce/WooCommerce.xcodeproj/project.pbxproj:5098]: This is so that CocoaPods 1.10 continues to use `inputFileList` and `outputFileList` features of Script Build Phases, because now CocoaPods 1.10 rightfully checks for this `compatibilityVersion` now to know if the feature is available (while it based that decision on the project's `objectVersion` before).

So if we kept `Xcode 8.0` as the project format / compatibilityVersion, CocoaPods would revert back to using the `inputFiles` and `outputFiles` property of Script Build Phases instead of taking advantage of `xcfilelists`, which would (1) not only make that diff way bigger (2) but also lose us the advantage of the project only referencing these `xcfilelist` – which means the `pbxproj` don't need to be changed and generate a diff if pods are changed and the script build phase requires to depend on different input/output files)

**Pull request merged**
